renku_notebooks/api/classes/image.py

```diff
@@ -41,16 +50,25 @@ def _get_docker_token(self, image: "Image") -> Optional[str]:
             # the request status code and header are not what is expected
             return None
         www_auth = WWWAuthenticate.from_header(auth_req.headers["Www-Authenticate"])
+        if not www_auth:
+            return None
         params = {**www_auth.parameters}
         realm = params.pop("realm")
+        if not realm:
+            return None
         headers = {"Accept": "application/json"}
         if self.oauth2_token:
             creds = base64.urlsafe_b64encode(f"oauth2:{self.oauth2_token}".encode()).decode()
             headers["Authorization"] = f"Basic {creds}"
         token_req = requests.get(realm, params=params, headers=headers)
-        return token_req.json().get("token")
+        return str(token_req.json().get("token"))
```

Comment:
Shouldn't we check if `....get("token")` returns `None` and return `None` if it's the case and convert to string otherwise?

Reply:
Good point. That is a bit too brave you are right.